### PR TITLE
tf: add success rate alarms for consumer and gateway

### PR DESCRIPTION
**Pull request by justIrina**

### CODE REVIEW

File: deployments/terraform/modules/consumer/alarms.tf [hunk 1]
```diff
@@ -0,0 +1,46 @@
+resource "aws_cloudwatch_metric_alarm" "success-rate" {
```
Comment (j4k4) [inline deployments/terraform/modules/consumer/alarms.tf:1]: the module names are quite generic right know.  
we also have to consider, that there are different consumers than sqs.  
i would suggest "grouping" the alarms with an "alarm-" prefix and to include input type.  
eg: `alarm_consumer_sqs`

File: deployments/terraform/modules/consumer/variables.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
+variable "project" {}
+variable "environment" {}
+variable "family" {}
+variable "application" {}
+
+variable "alarm_create" {
```
Comment (j4k4) [inline deployments/terraform/modules/consumer/variables.tf:6]: if the module is alarm context only, we can skip the alarm prefix for the variables i guess.

File: deployments/terraform/modules/consumer/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  description = "Required percentage of successful requests"
+}
+
+variable "alarm_period" {
+  type        = number
+  default     = 300
+  description = "The period in seconds over which the specified statistic is applied"
+}
+
+variable "model" {
```
Comment (j4k4) [inline deployments/terraform/modules/consumer/variables.tf:36]: i would move the model variable up top and group it together with the other application identifiers as they are all identifying the "source" of this alarm.

File: deployments/terraform/modules/gateway/alarms.tf [hunk 4]
```diff
@@ -0,0 +1,46 @@
+resource "aws_cloudwatch_metric_alarm" "success-rate" {
```
Comment (j4k4) [inline deployments/terraform/modules/gateway/alarms.tf:1]: i would suggest module name: `alarm_gateway`

File: deployments/terraform/modules/gateway/alarms.tf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  metric_query {
+    id          = "requests"
+    return_data = false
+
+    metric {
+      dimensions = {
+        "path" = var.api_request_path
+      }
+      metric_name = "ApiRequestCount"
```
Comment (j4k4) [inline deployments/terraform/modules/gateway/alarms.tf:18]: i tend to go with the request count from the load balancer, as we would list all paths from a gateway here as we have quite some gateways with multiple paths.

**Review by j4k4 — CHANGES_REQUESTED** [review]

File: deployments/terraform/modules/alarm_gateway/variables.tf [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  description = "Required percentage of successful requests"
+}
+
+variable "period" {
+  type        = number
+  default     = 300
+  description = "The period in seconds over which the specified statistic is applied"
+}
+
+variable "api_request_path" {
```
Comment (j4k4) [inline deployments/terraform/modules/alarm_gateway/variables.tf:36]: this one isn't need anymore, right?

File: deployments/terraform/modules/alarm_gateway/variables.tf [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  type        = string
+  default     = null
+  description = "Api request path"
+}
+
+variable "aws_lb_application_arn_suffix" {
```
Comment (j4k4) [inline deployments/terraform/modules/alarm_gateway/variables.tf:42]: is it a suffix or do we provide the full arn?

Comment (justIrina) [reply deployments/terraform/modules/alarm_gateway/variables.tf:42]: according to https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudwatch_metric_alarm examples we have to provide suffix
also here https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html when load balancer dimenssion is used it isnt' and arn

Comment (j4k4) [reply deployments/terraform/modules/alarm_gateway/variables.tf:42]: okay, it's the "final portion of the load balancer ARN" :+1:

**Review by j4k4 — APPROVED** [review]

**Review by j4k4 — APPROVED** [review]
+1

**Pull request merged**